overlord/ifacestate/hotplug.go

```diff
@@ -173,3 +174,20 @@ func suggestedSlotName(devinfo *hotplug.HotplugDeviceInfo, fallbackName string)
 	}
 	return shortestName
 }
+
+// create tasks to disconnect slots of given device, update the slot in the repository, then connect it back.
+func updateDevice(st *state.State, ifaceName, hotplugKey string, newAttrs map[string]interface{}) *state.TaskSet {
+	hotplugDisconnect := st.NewTask("hotplug-disconnect", fmt.Sprintf("Disable connections of device %q", hotplugKey))
+	setHotplugAttrs(hotplugDisconnect, ifaceName, hotplugKey)
+
+	updateSlot := st.NewTask("hotplug-update-slot", fmt.Sprintf("Update slot of interface %s, hotplug key %q", ifaceName, hotplugKey))
```

Comment:
Nothing actionable yet but I think we need to see how this looks like in practice once we interact with hot plug and word-smith it into something sensible to the user.